### PR TITLE
Add lint erros and run on branch only

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-7073

## Description
After adding LOG and return_results errors to lint, in this pr I remerge the reverted commits and make lint validate this only when all_packs flag is False, indicating we are on a feature branch

## Testing:

Running Nightly Lint with this branch (should be equivalent to without this issue): https://code.pan.run/xsoar/content/-/pipelines/5938899 - Passed :) 

using the branch https://github.com/demisto/content/pull/28718 

Locally:
* Ran with -i and got relevant errors
* Ran with all_packs=True in command_builder and did not see errors

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,8 @@
 * Fixed an issue where **lint** command failed to pull docker image in Gitlab environment.
 * Fixed an issue in **doc-review** command where escape characters within Markdown files were detected as invalid words.
 * Fixed an issue where **validate** failed on infrastructure test files.
+* **lint** will now fail on `demisto.results` and `return_outputs` usage, when a pack is `xsoar` or `partner` supported.
```
Comment (dorschw) [inline CHANGELOG.md:38]: pull from master and sync the changelog please

File: demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py [hunk 2]
```diff
@@ -166,7 +166,6 @@ def leave_module(self, node):
     """
 
     # -------------------------------------------- Call Node ---------------------------------------------
-
```
Comment (eyalpalo) [inline demisto_sdk/commands/lint/resources/pylint_plugins/base_checker.py:169]: Please revert

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice!

File: demisto_sdk/commands/lint/commands_builder.py [hunk 3]
```diff
@@ -96,7 +102,10 @@ def build_bandit_command(files: List[Path]) -> str:
 
 
 def build_xsoar_linter_command(
-    files: List[Path], support_level: str = "base", formatting_script: bool = False
+    files: List[Path],
+    support_level: str = "base",
```
Comment (dorschw) [inline demisto_sdk/commands/lint/commands_builder.py:106]: Consider using `Literal` as the type hint here

**Review by eyalpalo — APPROVED** [review]

**Pull request merged**